Packs/ZeroNetworksSegment/Integrations/ZeroNetworksSegmentEventCollector/ZeroNetworksSegmentEventCollector.py

```diff
@@ -0,0 +1,378 @@
+        previous_ids (list): List of IDs from the previous fetch to track.
+
+    Returns:
+        Dict[str, Any]: Updated dictionary containing the last run details.
+    """
+    last_run[log_type] = {
+        "last_fetch": last_event_time,
+        "previous_ids": previous_ids
+    }
+    return last_run
+
+
+def create_id(event: dict, log_type: str) -> str:
+    """
+    Create a unique ID for an event based on its log type.
+
+    Args:
+        event (dict): Dictionary containing event details.
+        log_type (str): Type of log to determine how to generate the ID (e.g., audit or network activities).
+
+    Returns:
+        str: A unique ID generated for the event, represented as a SHA-256 hash.
+    """
+    timestamp = event.get("timestamp")
+    if log_type == AUDIT_TYPE:
+        reported_object_id = event.get("reportedObjectId", "")
+        performed_by_name = event.get("performed_by", {}).get("id", "")
+        combined_string = f"{timestamp}-{reported_object_id}-{performed_by_name}"
+
+    elif log_type == NETWORK_ACTIVITIES_TYPE:
+        src_asset_id = event.get("src", {}).get("assetId", "")
+        dst_asset_id = event.get("dst", {}).get("assetId", "")
+        combined_string = f"{timestamp}-{src_asset_id}-{dst_asset_id}"
+
+    hash_object = hashlib.sha256(combined_string.encode())
+    return hash_object.hexdigest()
+
+
+def process_events(events: list, previous_ids: list, last_event_time: int, max_results: int, num_results: int,
+                   log_type: str) -> tuple[list, list, int]:
+    """
+    Process a list of events to filter out new ones and update tracking information.
+
+    Args:
+        events (list): List of event dictionaries to process.
+        previous_ids (list): List of IDs from previously processed events.
+        last_event_time (int): Timestamp of the last event processed.
+        max_results (int): Maximum number of results to process.
+        num_results (int): Current number of results processed.
+        log_type (str): Type of log to determine how to process the events.
+
+    Returns:
+        Tuple[list, list, int]: A tuple containing:
+            - new_events (list): List of newly processed events.
+            - updated_previous_ids (list): Updated list of IDs from previously processed events.
+            - updated_last_event_time (int): Updated timestamp of the last event processed.
+    """
+    new_events = []
+    for event in events:
+        event_id = create_id(event, log_type)
+        if num_results == max_results:
+            break
+
+        if event_id not in previous_ids:
+            event['_TIME'] = timestamp_to_datestring(event.get('timestamp'), is_utc=True)
+            event['source_log_type'] = log_type
+            new_events.append(event)
+            event_timestamp = event.get("timestamp")
+            num_results += 1
+            if int(event_timestamp) > last_event_time:
+                demisto.debug('updating the last run')
```

Comment:
please remove this log it will potentialy happen for every id and will spam the logs.